### ISSUE TITLE
IPv6 stack completely disabled if ipv6 option is not set (breaking existing dualstack network)

### ISSUE THREAD

**Issue by bhuisgen** [moby/moby#32924]
**Description**

I have a dualstack network configuration and if I start the docker engine without the ipv6 flag, the IPv6 stack is totally disabled on the system, breaking my network.

If I don't want IPv6 on my containers why disabling all the IPv6 stack if ipv6 is not set ? In addition to this bad choice, if I set the flags ipv6=true + vip-cidr-ipv6 to fe80::1/64 for the default bridge, the daemon starts and disable all IPv6 ... In addition to breaking WLAN/LAN the localhost ::1 address is required for some network daemons like unbound DNS.

**Steps to reproduce the issue:**

1.use a  dualstack network with IPv6 + IPv4
2.start docker engine without ipv6 flag
````
/usr/bin/dockerd -H fd:// --storage-driver overlay2 --bip=172.30.0.1/24 --dns 8.8.8.8 --dns 8.8.4.4
````
3.ip addr show

**Describe the results you received:**

All network interfaces have their IPv6 addresses removed :

````
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc pfifo_fast state DOWN group default qlen 1000
    link/ether 68:f7:28:b2:d6:8a brd ff:ff:ff:ff:ff:ff
3: wlp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether ac:7b:a1:82:e0:d8 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.104/24 brd 192.168.0.255 scope global dynamic wlp4s0
       valid_lft 84432sec preferred_lft 84432sec
````

````
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.docker0.disable_ipv6 = 1
net.ipv6.conf.eth0.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
net.ipv6.conf.wlp4s0.disable_ipv6 = 1
````

**Describe the results you expected:**

To have my IPv4/IPv6 dualstack for WLAN/LAN working and nothing with IPv6 for my containers so engine must start without changing anything.

````
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc pfifo_fast state DOWN group default qlen 1000
    link/ether 68:f7:28:b2:d6:8a brd ff:ff:ff:ff:ff:ff
3: wlp4s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether ac:7b:a1:82:e0:d8 brd ff:ff:ff:ff:ff:ff
    inet 192.168.0.104/24 brd 192.168.0.255 scope global dynamic wlp4s0
       valid_lft 86396sec preferred_lft 86396sec
    inet6 2a01:cb11:391:d100:e136:224f:92d0:2907/64 scope global noprefixroute dynamic 
       valid_lft 897sec preferred_lft 297sec
    inet6 fe80::790d:8f97:be49:1e22/64 scope link 
       valid_lft forever preferred_lft forever
````

````
net.ipv6.conf.all.disable_ipv6 = 0
net.ipv6.conf.default.disable_ipv6 = 0
net.ipv6.conf.docker0.disable_ipv6 = 1
net.ipv6.conf.eth0.disable_ipv6 = 0
net.ipv6.conf.lo.disable_ipv6 = 0
net.ipv6.conf.wlp4s0.disable_ipv6 = 0
````

**Output of `docker version`:**

```
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 17:50:07 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 17:50:07 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 12
 Running: 1
 Paused: 0
 Stopped: 11
Images: 3893
Server Version: 17.04.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.0-2-amd64
Operating System: Debian GNU/Linux 9 (stretch)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.4GiB
Name: laptop
ID: D6YZ:RQ4R:GATU:BDEN:2BMU:YHSC:BUUD:VEI7:S252:OZ2J:4GTM:BDLQ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Same problem on production with dualstack EC2 instances.

**Comment by thaJeztah** [comment]
This may be a duplicate of https://github.com/moby/moby/issues/32433, could you check if your case is the same?

**Comment by bhuisgen** [comment]
I confirm @thaJeztah it's the same problem. The bug is related to using containers with `--net=host`.

If I remove my containers, start the engine without ipv6 (after first system boot), all interfaces have IPv6. If I restart the engine with ipv6 enabled + ipv6 cidr block, it works too. But if I start a container with host network mode, all IPv6 addresses have lost.

So the bug is related to using containers with `--net=host` and I need to stay on v17.03 at this time.

**Comment by thaJeztah** [comment]
Thanks for checking! The issue will be addressed in 17.05, which is due first week of May. Let me close this issue because it has been fixed on "master" and in the 17.05 branch, but thanks for reporting, and feel free to continue the conversation 👍
